crates/rpc/tests/rpc.rs

```diff
@@ -0,0 +1,474 @@
+// These tests need Beerus to run in the background, hence why they're hidden behind the following feature.
+#![cfg(feature = "integration-tests")]
+
+use beerus_core::config::DEFAULT_PORT;
+use cached::Cached;
+use std::ops::Deref;
+
+use cached::SizedCache;
+use reqwest::Url;
+use starknet::{
+    core::types::{
+        BlockId, BlockWithTxHashes, BlockWithTxs, DeclareTransaction,
+        DeployAccountTransaction, FieldElement, InvokeTransaction,
+        MaybePendingBlockWithTxHashes, MaybePendingBlockWithTxs,
+        MaybePendingTransactionReceipt, Transaction, TransactionReceipt,
+    },
+    providers::{
+        jsonrpc::HttpTransport, JsonRpcClient, Provider, ProviderError,
+    },
+};
+
+fn rpc_client() -> JsonRpcClient<HttpTransport> {
+    let rpc_url: Url = format!("http://localhost:{}", DEFAULT_PORT)
+        .parse()
+        .expect("Invalid RPC URL");
+    JsonRpcClient::new(HttpTransport::new(rpc_url))
+}
+
+struct TestClient {
+    inner: JsonRpcClient<HttpTransport>,
+    block_cache: SizedCache<u64, MaybePendingBlockWithTxs>,
```

Comment:
I'm trying to understand this cache mechanism and don't know how it's working, so please correct me if I'm wrong 😄 

In each test case, the Test Client (with a fresh cache) is created via the context function, and then the cache is populated in this context call. Since the context call is the only place the cache is touched, and the cache key is always different because the key gets decreased by one in a loop, and then the client is destroyed for each test. How does it work, and what is the benefit of caching here?

Reply:
Nice catch @LKozlowski , it seems like the scope of cache is a single test-case. I believe a test-suite-wise cache would be more efficient in terms of reducing number of (duplicated) RPC calls.

But as long as those are tests, I think it is totally fine to take a step back here.
@pierre-l please feel free to drop the caching thing, no one is pushing for it really. We can totally leave it for the next PR.